packages/packages/documents/src/store/__tests__/sync.test.ts

```diff
@@ -0,0 +1,282 @@
+		const currentDocument = storeState.documents.documents[ currentDocumentId ];
+
+		expect( currentDocumentId ).toBe( 2 );
+		expect( currentDocument ).toEqual( {
+			id: 2,
+			title: 'Document 2',
+			status: 'publish',
+			isModified: false,
+			isSaving: false,
+			isSavingDraft: false,
+			userCan: {
+				publish: true,
+			},
+		} );
+	} );
+
+	it( 'should sync saving state of a document on V1 load', () => {
+		// Arrange.
+		const mockDocument = makeMockDocument();
+
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				{
+					...mockDocument,
+					editor: {
+						...mockDocument.editor,
+						isSaving: true,
+					},
+				},
+			] ),
+		};
+
+		// Act.
+		dispatchV1ReadyEvent();
+
+		// Assert.
+		const storeState = store.getState();
+		const { currentDocumentId } = storeState.documents;
+		const currentDocument = storeState.documents.documents[ currentDocumentId ];
+
+		expect( currentDocument.isSaving ).toBe( true );
+	} );
+
+	it( 'should sync saving state of a document on save', () => {
+		// Arrange.
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				makeMockDocument(),
+			] ),
+		};
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		const getCurrentDocument = () => {
+			const storeState = store.getState();
+			const { currentDocumentId } = storeState.documents;
+
+			return storeState.documents.documents[ currentDocumentId ];
+		};
+
+		// Act.
+		dispatchCommandBefore( 'document/save/save' );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( true );
+		expect( getCurrentDocument().isSavingDraft ).toBe( false );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/save' );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( false );
+		expect( getCurrentDocument().isSavingDraft ).toBe( false );
+	} );
+
+	it( 'should sync draft saving state of a document on save', () => {
+		// Arrange.
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				makeMockDocument(),
+			] ),
+		};
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		const getCurrentDocument = () => {
+			const storeState = store.getState();
+			const { currentDocumentId } = storeState.documents;
+
+			return storeState.documents.documents[ currentDocumentId ];
+		};
+
+		// Act.
+		dispatchCommandBefore( 'document/save/save', {
+			status: 'autosave',
+		} );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( false );
+		expect( getCurrentDocument().isSavingDraft ).toBe( true );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/save', {
+			status: 'autosave',
+		} );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( false );
+		expect( getCurrentDocument().isSavingDraft ).toBe( false );
+	} );
+
+	it( 'should sync modified state of a document on V1 load', () => {
+		// Arrange.
+		const mockDocument = makeMockDocument();
+
+		extendedWindow.elementor = {
```

Comment:
Not sure that there's anything to do about this, but: :(

Reply:
Done